tests/system_metrics/test_system_metrics_logging.py

```diff
@@ -39,11 +39,17 @@ def test_manual_system_metrics_monitor():
     for name in expected_metrics_name:
         assert name in metrics
 
+    # Check the step is correctly logged.
+    metrics_history = mlflow.tracking.MlflowClient().get_metric_history(
+        run.info.run_id, "cpu_utilization_percentage"
+    )
+    assert metrics_history[-1].step > 0
```

Comment:
Ah, is it difficult to test for the actual step because we're not using a mock clock?

Reply:
it's a glitch in multi-threading, basically it's not guaranteed that the daemon thread is logging exactly at the interval we set, so our test will be flaky if we assert the absolute number of steps.